### ISSUE TITLE
💬 Feedback - Fragment Generation in graphql folder

### ISSUE THREAD

**Issue by robksawyer** [contentful/template-blog-webapp-nextjs#39]
# Template feedback

Can you elaborate on how the `.graphql` fragments were generated originally in the `lib/graphql` folder? I'm a bit confused where to find the fragment info when a content model gets updated. These will manually need to be updated in that case, correct?

**Comment by mattvanvoorst-contentful** [comment]
Hi Rob, thanks for reaching out!

The `lib/graphql` folder consists of files that generate the output for `lib/__generated`. The config for the codegen can be found in `codegen.ts`. The codegen can be ran with yarn, npm or similar with the `graphql-codegen:watch` or `graphql-codegen:generate` commands. For more info on these I'd recommend checking out the section on code generation in the README.

The naming and separation of the files in `lib/graphql` was decided on by our development team and generally serve a single purpose; describing a GraphQL query for one of the Content Models created in Contentful. We opted to abstract our GraphQL fields in a fragment, because the codegen will generate a TypeScript interface for each fragment which we can then use to type for example our properties in a React component. 

If we take for example the `AuthorFields` fragments from the `authorFields.graphql` file. This fragment is created on the `ComponentAuthor` model, which directly maps to the `component - Author` content model in Contentful.

### How it works now
So we define the model in Contentful:
<img width="734" alt="image" src="https://user-images.githubusercontent.com/104915050/214286938-2eabfe12-dcd9-4954-b7bd-eee5e4b7be3d.png">

Which after running the codegen, generates the following type in our GraphQL schema file:
```ts
"This model describes an author for an article [See type definition](https://app.contentful.com/spaces/nvxmqufxogry/content_types/componentAuthor)"
type ComponentAuthor implements Entry {
    avatar(locale: String, preview: Boolean): Asset
    contentfulMetadata: ContentfulMetadata!
    internalName(locale: String): String
    linkedFrom(allowedLocales: [String]): ComponentAuthorLinkingCollections
    name(locale: String): String
    sys: Sys!
}
```

Which we can then use to create a GraphQL query:
```ts
fragment AuthorFields on ComponentAuthor {
  __typename
  name
  avatar {
    ...ImageFields
  }
}
```

Which then generates the following TypeScript type through the codegen, which we could use to type a React component:
```gql
export type AuthorFieldsFragment = {
  __typename: "ComponentAuthor";
  name?: string | null;
  avatar?: ({ __typename?: "Asset" } & ImageFieldsFragment) | null;
};
```

### How you could expand it.
So lets say you wanted to add another property/field to the author content model.

You would take the following steps:
1. Create the field on the content model in Contentful
2. Run the GraphQL codegen locally, so the new file gets added to the local schema file
3. Add the property to the fragment in the `authorFields.graphql` file
4. Run the codegen again, this will re-generate the `AuthorFieldsFragment` type.

If these steps ran successfully, you should now be able to use the new field you added, in any of your React components.

The tl:dr; is: After adding a new field in the Contentful content model, you have to let your GraphQL queries in your local project know the property now exists through the codegen.

### Final tip
I can highly recommend the following app: https://www.contentful.com/marketplace/app/graphql-playground/. It's free to use and installs directly into your Contentful environment. It lets you play around with GraphQL queries, and also automagically generates documentation based on your content models.
